charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl

```diff
@@ -57,31 +57,28 @@ Usage:
 
 
 {{/*
-Return the proper Docker Image Registry Secret Names evaluating values as templates
-{{ include "elasticsearch-exporter.image.pullSecret.name" ( dict "images" (list .Values.path.to.the.image1, .Values.path.to.the.image2) "context" $) }}
-*/}}
-{{- define "elasticsearch-exporter.image.pullSecret.name" -}}
-  {{- $pullSecrets := list }}
-  {{- $context := .context }}
+To help compatibility with other charts which use global.imagePullSecrets.
+Allow either an array of {name: pullSecret} maps (k8s-style), or an array of strings (more common helm-style).
+global:
+  imagePullSecrets:
+  - name: pullSecret1
+  - name: pullSecret2
 
-  {{- if $context.Values.global }}
-    {{- range $context.Values.global.imagePullSecrets -}}
-      {{/* Is plain array of strings, compatible with all bitnami charts */}}
-      {{- $pullSecrets = append $pullSecrets (include "elasticsearch-exporter.tplvalue.render" (dict "value" . "context" $context)) -}}
-    {{- end -}}
-  {{- end -}}
-  {{- range .images -}}
-    {{- if .pullSecret -}}
-      {{- $pullSecrets = append $pullSecrets (include "elasticsearch-exporter.tplvalue.render" (dict "value" .pullSecret "context" $context)) -}}
-    {{- end -}}
-  {{- end -}}
+or
 
-  {{- if (not (empty $pullSecrets)) }}
-imagePullSecrets:
-    {{- range $pullSecrets | uniq }}
-  - name: {{ . }}
-    {{- end }}
+global:
+  imagePullSecrets:
+  - pullSecret1
+  - pullSecret2
+*/}}
+{{- define "elasticsearch-exporter.imagePullSecrets" -}}
+{{- range .Values.global.imagePullSecrets }}
+  {{- if eq (typeOf .) "map[string]interface {}" }}
+- {{ toYaml . | trim }}
+  {{- else }}
+- name: {{ . }}
```

Comment:
The current helper template supports templating. I would prefer not to remove this functionality as removing it might break some installations. My suggestion would be

```suggestion
  {{- if eq (typeOf .) "map[string]interface {}" }}
- {{ tpl (toYaml .) $ | trim }}
  {{- else }}
- name: {{ tpl . $ }}
```

Please, add a comment line to the field in values.yaml on its supported types and templating.